### PR TITLE
Krita 5.0.8

### PR DESCRIPTION
**Pull request by vulongm**
- Update dependencies and patches supersedes #67
- Add ``--talk-name=com.canonical.AppMenu.Registrar`` solving #68
- Fix PyQt5 build
- Add f-e-d-c to gmic

### CODE REVIEW

**Comment by travier** [comment]
bot, build

File: patches/krita/AppData.patch [hunk 1]
```diff
@@ -33,8 +33,8 @@ index d02fb68..62ba534 100644
    </provides>
    <content_rating type="oars-1.1"/>
    <releases>
--    <release date="2022-04-27" version="5.0.6">
-+    <release date="2022-04-27" version="5.0.6"/>
+-    <release date="2022-05-25" version="5.0.8">
```
Comment (travier) [inline patches/krita/AppData.patch:36]: This one should add the entry, not replace the existing one. We should also push that upstream.

File: org.kde.krita.yaml [hunk 2]
```diff
@@ -865,10 +878,8 @@ modules:
           project-id: 1477
           stable-only: true
           url-template: https://s3.amazonaws.com/json-c_releases/releases/json-c-$version.tar.gz
+  - shared-modules/intltool/intltool-0.51.json
   - name: libmypaint
-    buildsystem: cmake-ninja
```
Comment (travier) [inline org.kde.krita.yaml:869]: Any reason we're not using cmake anymore here?

File: org.kde.krita.yaml [hunk 3]
```diff
@@ -645,50 +655,53 @@ modules:
...
+        url: https://files.pythonhosted.org/packages/39/5f/fd9384fdcb9cd0388088899c110838007f49f5da1dd1ef6749bfb728a5da/PyQt5_sip-12.11.0.tar.gz
+        sha256: b4710fd85b57edef716cc55fae45bfd5bfac6fc7ba91036f1dcc3f331ca0eb39
         x-checker-data:
           type: pypi
           name: PyQt5_sip
   - name: pyqt5
     buildsystem: simple
     build-commands:
-      - pip3 install --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "PyQt5" --no-build-isolation
+      - sip-install --confirm-license --no-designer-plugin --no-qml-plugin --no-dbus-python --no-tools --verbose --target-dir ${FLATPAK_DEST}/lib/python3.9/site-packages
```
Comment (travier) [inline org.kde.krita.yaml:698]: Is this intentional?

**Review by travier — CHANGES_REQUESTED** [review]
Mostly LGTM. Some comments.

**Comment by travier** [comment]
And thanks for working on this

File: org.kde.krita.yaml (hunk 3 above)
Comment (vulongm) [reply org.kde.krita.yaml:698]: Yes, I noticed the PyQt build was failing earlier as pip install provided a license prompt which the builder couldn't accept.

File: patches/krita/AppData.patch (hunk 1 above)
Comment (vulongm) [reply patches/krita/AppData.patch:36]: Thanks for the catch, going to fix that patch. Looks like the 5.1 builds coming up next don't have the missing slash, but sent a PR earlier for the 5.0 branch.

File: org.kde.krita.yaml (hunk 2 above)
Comment (hfiguiere) [reply org.kde.krita.yaml:869]: it's faster? the rule is if cmake-ninja work, use it. (it's `cmake -G Ninja`)

Comment (vulongm) [reply org.kde.krita.yaml:869]: libmypaint doesn't support building using cmake without the custom patches that Krita added here https://invent.kde.org/graphics/krita/-/tree/master/3rdparty/ext_mypaint. Would it be better to build it without those patches or keep the patches in to use cmake-ninja?

File: patches/krita/AppData.patch (hunk 1 above)
Comment (vulongm) [reply patches/krita/AppData.patch:36]: @travier I do notice Krita has multiple different AppData files. The one on their main branch has multiple versions. https://invent.kde.org/graphics/krita/-/blob/master/krita/org.kde.krita.appdata.xml#L560 but under the branch for each version it seems to only contain the latest release for that branch. https://invent.kde.org/graphics/krita/-/blob/krita/5.1/krita/org.kde.krita.appdata.xml#L516.

File: org.kde.krita.yaml (hunk 2 above)
Comment (travier) [reply org.kde.krita.yaml:869]: Hum, there is unfortunately no good options here:
- If we don't use the patches then we deviate from how upstream Krita tests & builds
- If we use the patches, we deviate from how upstream libmypaint tests & builds
Given that Krita is already a snowflake, I'd say we keep doing as they do upstream?
WDYT?

File: patches/krita/AppData.patch (hunk 1 above)
Comment (travier) [reply patches/krita/AppData.patch:36]: Well they should not do that AFAIK. Each version should include all of the previous ones. Can you make an upstream bug report and/or the send a PR?

File: org.kde.krita.yaml (hunk 2 above)
Comment (vulongm) [reply org.kde.krita.yaml]: Sure, going ahead and readding the patches for cmake-ninja.

File: patches/krita/AppData.patch (hunk 1 above)
Comment (vulongm) [reply patches/krita/AppData.patch:36]: Is there a link to a documentation page that mentions to include prior version that can be used as reference for a PR/bug report?

Comment (travier) [reply patches/krita/AppData.patch:36]: https://freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-releases. It's not mandatory, but preferred.

**Review by travier — APPROVED** [review]
LGTM. The appdata fixes can come later. Let me know when you've tested it and I can merge it. Thanks!

**Comment by vulongm** [comment]
> LGTM. The appdata fixes can come later. Let me know when you've tested it and I can merge it. Thanks!

@travier Works here testing it out!

**Pull request merged**